crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -909,33 +909,42 @@ impl<'src> Parser<'src> {
         }
     }
 
+    /// Parses a boolean operation expression.
+    ///
+    /// Note that the boolean `not` operator is parsed as a unary operator and
+    /// not as a boolean operation.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at a `or` or `and` token.
+    ///
+    /// See: <https://docs.python.org/3/reference/expressions.html#boolean-operations>
     fn parse_bool_operation_expression(
         &mut self,
         lhs: Expr,
         start: TextSize,
-        op: TokenKind,
-        op_bp: Precedence,
+        operator_token: TokenKind,
+        operator_binding_power: Precedence,
     ) -> ast::ExprBoolOp {
         let mut values = vec![lhs];
         let mut progress = ParserProgress::default();
 
-        // Keep adding `expr` to `values` until we see a different
-        // boolean operation than `op`.
+        // Keep adding the expression to `values` until we see a different
+        // boolean operation than `operator_token`.
```

Comment:
Nit: We stop at any token that's not the same operator. 

```suggestion
        // token than `operator_token`.
```